crates/kitsune_p2p/transport_quic/src/listener.rs

```diff
@@ -1,26 +1,114 @@
-use futures::{future::FutureExt, stream::StreamExt};
+use futures::{future::FutureExt, sink::SinkExt, stream::StreamExt};
 use kitsune_p2p_types::{
-    dependencies::{ghost_actor, url2::*},
-    transport::transport_connection::*,
-    transport::transport_listener::*,
+    dependencies::{ghost_actor, ghost_actor::GhostControlSender, url2::*},
     transport::*,
 };
-use std::net::SocketAddr;
+use std::{collections::HashMap, net::SocketAddr};
+
+/// Convert quinn async read/write streams into Vec<u8> senders / receivers.
+fn tx_bi_chan(
```

Comment:
Is the point of this function to basically map the quinn streams to futures channels?